src/Contracts/User.php

```diff
@@ -38,4 +38,18 @@ public function getEmail();
      * @return string|null
      */
     public function getAvatar();
+
+    /**
+     * Get the raw user array.
+     *
+     * @return array
+     */
+    public function getRaw();
+
+    /**
+     * Get the user's attributes.
+     *
+     * @return array
+     */
+    public function getAttributes();
```

Comment:
We can't add methods to interfaces on patch releases. This will break all apps implementing custom providers.

Reply:
Of course, I will revert the interface. Is there a major release branch that these additions can go on?

Reply:
Maybe we can add it as `@method` annotations for now?

Reply:
Yeah, I considered that. Seems to do the trick on my local branch. The whole PR came about as a step toward avoiding direct type-hinting with inline PHPDocs, due to PHPStan not recognizing the public properties:

```php
/** @var \Laravel\Socialite\Two\User $user */
$user = Socialite::driver('facebook')->user();

$attributes = $user->attributes;
```

With the proposed changes it would simply be:

```php
$user = Socialite::driver('facebook')->user();

$attributes = $user->getAttributes();
```

This proposed change leverages new getters like the proposed `getAttributes()` method, which is a more explicit and likely better practice for accessing the user’s attributes. As a method, it could also be added to the class PHPDoc as you had suggested. This would make for better static analysis support, since properties aren't meant for interfaces. The change was aimed at making PHPStan recognize the attributes correctly without relying on the type hint.

However, I can see that @taylorotwell doesn't think that is needed. So, this is all a moot issue now. I'll just revert my development back to type-hinting the `$user` var to make PHPStan happy.

Thanks guys.